lib/ace/edit_session.js

```diff
@@ -1116,6 +1118,7 @@ var EditSession = function(text, mode) {
      *
      **/
     this.insert = function(position, text) {
+        this.$bidiHandler.markAsDirty();
```

Comment:
sorry, i meant line 255 https://github.com/ashensis/ace/blob/8b21722f4dff1e43a5c4debf066a54b0bcf6a9d3/lib/ace/edit_session.js#L255, session can be changed without calls to insert/remove. The only reliable way to handle that is to listen to the change event.